src/java/picard/sam/MergeBamAlignment.java

```diff
@@ -146,11 +146,11 @@
 
     @Option(doc = "Reserved alignment attributes (tags starting with X, Y, or Z) that should be " +
             "brought over from the alignment data when merging.")
-    public List<String> ATTRIBUTES_TO_RETAIN = new ArrayList<String>();
+    public List<String> ATTRIBUTES_TO_RETAIN = new ArrayList<>();
 
     @Option(doc = "Attributes from the alignment record that should be removed when merging." +
             "  This overrides ATTRIBUTES_TO_RETAIN if they share common tags.")
-    public List<String> ATTRIBUTES_TO_REMOVE = new ArrayList<String>();
+    public List<String> ATTRIBUTES_TO_REMOVE = new ArrayList<>();
```

Comment:
Some comments are optional.  I tend to like separate commits for a clean history.  To each his/her own.